### PR TITLE
Get `yarn markdown-lint` to work

### PR DESCRIPTION
**Pull request by ptgott**
On `gravitational/docs`, the `yarn markdown-lint` script runs the docs site's `remark`-based linters on the Teleport documentation content. This change ensures that the script works as expected on Docusaurus content.

- Add dependencies for `yarn markdown-lint`.
- Add a `build-remark` script for building remark plugins in order to run the linter.
- Copy `.remarkrc.mjs` from `gravitational/docs`, but add an `updatePaths` function to remark-includes to work with the plugin's new interface.
- Edit `getVersionFromFile` to accommodate linting. Get a version if the file is currently in the `content` directory, which we need to do in order to get `remarkIncludes` to work when linting files in `content`. Also add branching in `getCurrentDir` to work with the structure of the `content` directory.
- Update package.json to use the versions of `remark` packages found in gravitational/docs. This minimizes the content changes we'll need to make to accommodate the linting configuration.

### CODE REVIEW

**Comment by taraspos** [comment]
@ptgott seems like you need to review some dependencies

File: package.json [hunk 1]
```diff
@@ -77,15 +78,26 @@
...
     "tsm": "^2.3.0",
     "typescript": "~5.5.2",
     "unified": "^11.0.5",
     "unified-lint-rule": "^3.0.0",
     "unist": "^0.0.1",
-    "unist-util-find": "^3.0.0",
-    "unist-util-visit-parents": "^6.0.1",
+    "unist-util-find": "^1.0.2",
+    "unist-util-visit": "4.1.0",
+    "unist-util-visit-parents": "^5.1.0",
```
Comment (taraspos) [inline package.json:100]: Seems like some dependencies got downgraded

File: package.json [hunk 2]
```diff
@@ -45,12 +46,12 @@
     "@mdx-js/react": "^3.0.0",
     "clsx": "^2.1.1",
     "date-fns": "^3.6.0",
+    "highlightjs-terraform": "https://github.com/highlightjs/highlightjs-terraform#eb1b9661e143a43dff6b58b391128ce5cdad31d4",
```
Comment (taraspos) [inline package.json:49]: Any reason why this is needed?

Comment (ptgott) [reply package.json:49]: We used this to enable HCL syntax highlighting in the previous docs site. We're using Docusaurus-native syntax highlighting now, so we shouldn't need this until (potentially) we address #6

File: package.json (hunk 1 above)
Comment (ptgott) [reply package.json:100]: I was trying to replicate our `gravitational/docs` configuration to minimize new linting errors. I had to fix new linting errors anyway, so I've reverted these downgrades

**Review by taraspos — APPROVED** [review]

**Pull request merged**